fuzz/fuzz_targets/cranelift-fuzzgen.rs

```diff
@@ -231,27 +240,40 @@ impl TestCase {
         let inputs = gen.generate_test_inputs(&main.signature)?;
 
         Ok(TestCase {
-            isa,
+            isa_builder: Rc::new(isa_builder),
+            flags_builder,
             functions,
             ctrl_planes,
             inputs,
             compare_against_host,
         })
     }
 
-    fn to_optimized(&self) -> Self {
+    fn build_isa(&self, fuel: Option<u8>) -> anyhow::Result<OwnedTargetIsa> {
+        use anyhow::Context;
+        let mut flags_builder = self.flags_builder.clone();
+        flags_builder
+            .set("control_plane_fuel", &fuel.unwrap_or_default().to_string())
+            .context("failed to set control_plane_fuel flag")?;
+        self.isa_builder
+            .finish(settings::Flags::new(flags_builder))
+            .context("failed to build isa")
+    }
+
+    fn to_optimized(&self, isa: &dyn TargetIsa) -> Self {
         let optimized_functions: Vec<Function> = self
             .functions
             .iter()
             .map(|func| {
                 let mut ctx = Context::for_function(func.clone());
-                ctx.optimize(self.isa.as_ref()).unwrap();
+                ctx.optimize(isa).unwrap();
                 ctx.func
             })
             .collect();
 
         TestCase {
-            isa: self.isa.clone(),
+            isa_builder: self.isa_builder.clone(),
```

Comment:
This is the reason I wrapped the `isa_builder` in an `Rc`. `to_optimized` takes a reference and returns an owned `Self`, so the `isa_builder` must either be cloned or shared.

I tried to make the `isa_builder` `Clone`, for that I would have to make the following also `Clone`:
- [`CodegenError`](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/codegen/src/result.rs#L12)
- [`RegisterMappingError`](https://github.com/bytecodealliance/wasmtime/blob/main/cranelift/codegen/src/isa/unwind/systemv.rs#L18)

If making these two `Clone` is better than using `Rc`, I can change that.